src/test/java/uk/gov/pay/products/resources/ProductResourceIT.java

```diff
@@ -958,4 +961,77 @@ public void shouldReturnMetadata_whenMetadataExistsForAProduct() {
                 .body(METADATA + ".key",is("value"))
                 .body(METADATA + ".secondkey",is("value2"));
     }
+
+    @Test
+    public void shouldReturnUsageStats_whenProductAndPaymensExist() {
+        String nextUrl = "www.gov.uk/pay";
+
+        String productExternalId = randomUuid();
+        String productExternalId2 = randomUuid();
+        String paymentExternalId1 = randomUuid();
+        String paymentExternalId2 = randomUuid();
+
+        ProductEntity productEntity = ProductEntityFixture.aProductEntity()
+                .withGatewayAccountId(1)
+                .withExternalId(productExternalId)
+                .withType(ProductType.ADHOC)
+                .build();
+        ProductEntity productEntity2 = ProductEntityFixture.aProductEntity()
+                .withGatewayAccountId(2)
+                .withExternalId(productExternalId2)
+                .withType(ProductType.ADHOC)
+                .build();
+
+        Product product = productEntity.toProduct();
+        Product product2 = productEntity2.toProduct();
+
+        databaseHelper.addProduct(product);
+        databaseHelper.addProduct(product2);
+
+        productEntity.setId(databaseHelper.findProductId(productExternalId));
+        productEntity2.setId(databaseHelper.findProductId(productExternalId2));
+
+        PaymentEntity payment1 = aPaymentEntity()
+                .withExternalId(paymentExternalId1)
+                .withProduct(productEntity)
+                .withNextUrl(nextUrl)
+                .withReferenceNumber(paymentExternalId1.substring(0, 9))
+                .build();
+
+        databaseHelper.addPayment(payment1.toPayment(), 1);
+
+        PaymentEntity payment2 = aPaymentEntity()
+                .withExternalId(paymentExternalId2)
+                .withProduct(productEntity2)
+                .withNextUrl(nextUrl)
+                .withReferenceNumber(paymentExternalId2.substring(0, 9))
+                .build();
+
+        databaseHelper.addPayment(payment2.toPayment(), 2);
+
+        ValidatableResponse response =  givenSetup()
+                .when()
+                .accept(APPLICATION_JSON)
+                .get(format("/v1/api/stats/products"))
+                .then()
+                .statusCode(200);
+
+        response
+                .body("size", is(2))
+                .body("[0].payment_count", is(1))
+                .body("[0].product.external_id", is(product.getExternalId()))
```

Comment:
Can we assert on all fields that we will be expected ? We are not asserting on date/product fields in DAOIT or here

Reply:
The product field has an assertion that the correct product is returned ^ will add one for dates